python/ray/data/read_api.py

```diff
@@ -205,9 +205,14 @@ def read_datasource(
     Returns:
         Dataset holding the data read from the datasource.
     """
-
     # TODO(ekl) remove this feature flag.
-    if "RAY_DATASET_FORCE_LOCAL_METADATA" in os.environ:
+    force_local = "RAY_DATASET_FORCE_LOCAL_METADATA" in os.environ
+    pa = _lazy_import_pyarrow()
+    if pa:
+        partitioning = read_args.get("dataset_kwargs", {}).get("partitioning", None)
```

Comment:
Are all of them non-serializable or just the Hive one? Also, is this a known upstream issue?

I'd favor trying to serialize it and falling back, rather than assuming they're all non serializable. That would make the code more general/future-proof.